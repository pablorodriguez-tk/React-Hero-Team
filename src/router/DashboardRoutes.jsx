import React from 'react';
import { Redirect, Route, Switch } from 'react-router';
import { Navbar } from '../components/ui/Navbar';
import { HeroDetailScreen } from '../pages/hero/HeroDetailScreen';
import { HeroScreen } from '../pages/hero/HeroScreen';
import { SearchScreen } from '../pages/hero/SearchScreen';

const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-2">
        <Switch>
          <Route exact path="/heroteam" component={HeroScreen} />
          <Route exact path="/hero/:heroeId" component={HeroDetailScreen} />
          <Route exact path="/search" component={SearchScreen} />
          <Redirect to="/heroteam" />
        </Switch>
      </div>
    </>
  );
};

export default DashboardRoutes;
