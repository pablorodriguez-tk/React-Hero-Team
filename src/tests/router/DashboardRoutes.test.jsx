import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router';
import DashboardRoutes from '../../router/DashboardRoutes';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Tests on DashboardRoutes', () => {
  test('Should be displayed correctly', () => {
    const initState = {
      auth: {
        isLoggedIn: true,
        email: 'challenge@alkemy.org',
      },
      heroes: {
        heroTeam: [],
      },
      _persist: {
        version: -1,
        rehydrated: true,
      },
    };

    const store = mockStore(initState);

    const wrapper = render(
      <Provider store={store}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </Provider>
    );

    const logout = wrapper.queryByText('Logout');
    expect(logout).toBeInTheDocument();
  });
});
