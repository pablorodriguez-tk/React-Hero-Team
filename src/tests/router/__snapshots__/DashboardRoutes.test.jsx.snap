// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on DashboardRoutes Should be displayed correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="/heroteam"
          >
            Hero Team
          </a>
          <div
            class="navbar-collapse"
          >
            <div
              class="navbar-nav"
            >
              <a
                class="nav-item nav-link"
                href="/search"
              >
                Search
              </a>
            </div>
          </div>
          <ul
            class="navbar-nav ml-auto"
          >
            <button
              class="nav-item nav-link btn"
              data-testid="logoutButton"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
      <div
        class="container mt-2"
      >
        <h1
          class="d-flex justify-content-center animate__animated animate__fadeIn"
          data-testid="heroScreen"
        >
          My hero team
        </h1>
        <div
          class="animate__animated animate__fadeIn"
        >
          <h2
            class="d-flex justify-content-center "
          >
            Type: 
            <b />
          </h2>
          <hr />
          <div
            class="row"
          >
            <div
              class="col-6 text-center"
            >
              <h2>
                Average powerstats
              </h2>
              <ul
                class="list-group list-group-flush"
              >
                <li
                  class="list-group-item"
                >
                  Combat: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Durability: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Intelligence: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Power: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Speed: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Strength: 
                  <b>
                    0.0
                  </b>
                </li>
              </ul>
            </div>
            <div
              class="col-6 text-center"
            >
              <h2>
                Average height and weight
              </h2>
              <ul
                class="list-group list-group-flush"
              >
                <li
                  class="list-group-item"
                >
                  Height: 
                  <b>
                    0.0
                     cm
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Weight: 
                  <b>
                    0.0
                     Kg
                  </b>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="animate__animated animate__fadeIn"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/heroteam"
        >
          Hero Team
        </a>
        <div
          class="navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-item nav-link"
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <ul
          class="navbar-nav ml-auto"
        >
          <button
            class="nav-item nav-link btn"
            data-testid="logoutButton"
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
    <div
      class="container mt-2"
    >
      <h1
        class="d-flex justify-content-center animate__animated animate__fadeIn"
        data-testid="heroScreen"
      >
        My hero team
      </h1>
      <div
        class="animate__animated animate__fadeIn"
      >
        <h2
          class="d-flex justify-content-center "
        >
          Type: 
          <b />
        </h2>
        <hr />
        <div
          class="row"
        >
          <div
            class="col-6 text-center"
          >
            <h2>
              Average powerstats
            </h2>
            <ul
              class="list-group list-group-flush"
            >
              <li
                class="list-group-item"
              >
                Combat: 
                <b>
                  0.0
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Durability: 
                <b>
                  0.0
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Intelligence: 
                <b>
                  0.0
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Power: 
                <b>
                  0.0
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Speed: 
                <b>
                  0.0
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Strength: 
                <b>
                  0.0
                </b>
              </li>
            </ul>
          </div>
          <div
            class="col-6 text-center"
          >
            <h2>
              Average height and weight
            </h2>
            <ul
              class="list-group list-group-flush"
            >
              <li
                class="list-group-item"
              >
                Height: 
                <b>
                  0.0
                   cm
                </b>
              </li>
              <li
                class="list-group-item"
              >
                Weight: 
                <b>
                  0.0
                   Kg
                </b>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="animate__animated animate__fadeIn"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
